### PR TITLE
CLN: _wrap_applied_output

### PR DESCRIPTION
**Pull request by rhshadrach**
- [ ] closes #xxxx
- [ ] tests added / passed
- [x] passes `black pandas`
- [x] passes `git diff upstream/master -u -- "*.py" | flake8 --diff`
- [ ] whatsnew entry

### CODE REVIEW

File: pandas/core/groupby/generic.py [hunk 1]
```diff
@@ -1203,16 +1203,27 @@ def _wrap_applied_output(self, keys, values, not_indexed_same=False):
...
             # self._selection_name not passed through to Series as the
             # result should not take the name of original selection
             # of columns
-            return self.obj._constructor_sliced(values, index=key_index)
+            if self.as_index:
+                return self.obj._constructor_sliced(values, index=key_index)
+            else:
+                result = DataFrame(values, index=key_index, columns=[self._selection])
+                self._insert_inaxis_grouper_inplace(result)
+                return result
```
Comment (jbrockmendel) [inline pandas/core/groupby/generic.py:1224]: looks like this line isnt covered (the equivalent line isnt covered in master, so not necessarily a problem for this PR).  any chance it is unreachable and can be removed?

**Comment by jbrockmendel** [comment]
Changing `v` to `first_not_none` is a clear improvement.  the other rearrangement im fine either way

File: pandas/core/groupby/generic.py (hunk 1 above)
Comment (rhshadrach) [reply pandas/core/groupby/generic.py:1224]: Added test to cover it.

**Comment by rhshadrach** [comment]
@jbrockmendel: Thanks for the review. Regarding the rearrangement, I thought it was better to separate "not a series" and "is a series" cases. Before they were (needlessly IMO) intertwined.

**Comment by jreback** [comment]
thanks @rhshadrach yeah groupby has built up over the years to be long / complicated functions, thanks for working on this.

**Pull request merged**